### PR TITLE
ESP-IDF Component Continuous Integration

### LINKED ISSUE #227: Add to ESP Component Registry
What do you think about adding libcanard to the [ESP Component Registry](https://components.espressif.com/)?
Since the ESP32 has great CAN bus support, having libcanard packaged in an ESP component would make it more accessible to developers.

Doing this would probably entail:
1. Requesting an OpenCyphal namespace through the [ESP Registry permissions](https://components.espressif.com/settings/permissions/) page. Giving key contributors access to this namespace.

2. Packaging up libcanard by following the [Packaging ESP-IDF Components](https://docs.espressif.com/projects/idf-component-manager/en/latest/guides/packaging_components.html) guide. This requires adding some metadata and build files.

3. Uploading the package to the ESP registry each time there's an update.

### PR DESCRIPTION
**Pull request by manforowicz**
This pull request is meant to resolve #227 by adding continuous integration for publishing Libcanard to the [ESP Component Registry](https://components.espressif.com/).

Files added:
- `idf_component.yml`. This file contains required ESP component metadata.
- `Kconfig`. ESP components use this file as a standard way for controlling build options. I've added options for enabling/disabling assertions and the CRC table.
- `CMakeLists.txt`. This file is required to register the component with the ESP build system.
- `.github/workflows/esp_publish.yml`. This workflow uploads the component to the ESP-IDF registry whenever a new GitHub release is published.
- `.github/workflows/esp_dry_run.yml`. This workflow does a dry-run of uploading to the ESP-IDF registry, without publishing anything. It can only be manually triggered. I've tested this in my repository, and it seems to work.

The GitHub workflows require `IDF_COMPONENT_API_TOKEN` with an [ESP registry token](https://components.espressif.com/settings/tokens/) to be added to the repository secrets.

### CODE REVIEW

File: CMakeLists.txt
Comment (pavel-kirienko) [inline CMakeLists.txt:1]: Is it possible to hide `CMakeLists.txt`, `Kconfig`, and `idf_component.yml` into a subdirectory, or at least the first two? I don't want to pollute the project root with third-party items.

File: idf_component.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Metadata for using libcanard as an ESP-IDF component.
+
+# Note: Version is not specified in this file,
+# because the 'esp_publish.yml' GitHub action
+# automatically sets it based on the release version.
+
+description: "OpenCyphal for embedded systems."
```
Comment (pavel-kirienko) [inline idf_component.yml:7]:
```suggestion
description: "Cyphal/CAN for embedded systems."
```

Cyphal comes with various transports, CAN is only one of them.

**Review by pavel-kirienko — CHANGES_REQUESTED** [review]
Thank you. I have created the token and added it to the repository secrets.

File: CMakeLists.txt
Comment (pavel-kirienko) [reply CMakeLists.txt:1]: This appears to be doable via [`repository_info`](https://docs.espressif.com/projects/idf-component-manager/en/latest/reference/manifest_file.html#repository-info)

**Comment by manforowicz** [comment]
As you suggested, to avoid polluting the root of the repository with ESP-related files, I tried using [`repository_info`](https://docs.espressif.com/projects/idf-component-manager/en/latest/reference/manifest_file.html#repository-info) and putting the ESP-related files into a subdirectory. However, this didn't work because ESP-IDF gets confused whenever the source code files are not in the directory where `CMakeLists.txt` is.

So I decided it would be cleanest to just have the continuous integration script copy all the required files into a temporary component directory before publishing. I tested the `esp_dry_run.yml` workflow locally, and it seems to work correctly.

File: .github/workflows/package_esp_component.sh
Comment (pavel-kirienko) [inline .github/workflows/package_esp_component.sh:1]: Shouldn't the script go into `/esp_metadata`?

File: esp_metadata/Kconfig [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+config CANARD_CRC_TABLE
+    bool "Enable libcanard CRC table"
+    default y
+    help
+        Set to 'n' to use slow but ROM-efficient transfer-CRC computation algorithm.
+        Doing so is expected to save ca. 500 bytes of ROM and
+        increase the cost of RX/TX transfer processing by ~half.
+
+
+endmenu
```
Comment (pavel-kirienko) [inline esp_metadata/Kconfig:25]: Missing newline at the end

**Review by pavel-kirienko — DISMISSED** [review]
Looks good. Please address the last two nits and we'll merge it.

**Review by pavel-kirienko — APPROVED** [review]

**Pull request merged**